### DISCUSSION TITLE
Seeking Guidance on Integrating ML/Genetic Algorithms with PyFluent for Nuclear Engineering Dissertation

### DISCUSSION THREAD

**Discussion by HBurgess173** [ansys/pyansys#332] [General]
Hello everyone,

Good afternoon! I am currently in the process of formulating a topic for my upcoming dissertation and am seeking some guidance and insights from this community.

A bit about me: I have a background in nuclear mechanical engineering and have been part of a degree apprenticeship program since 2019. Over the years, I've gained proficiency in various modules such as CFD, FEA, Engineering Mathematics, MATLAB, Thermofluidic Systems, Stress, and Dynamics. My expertise also extends to nuclear-centric subjects like reactor physics, reactor chemistry, and reactor materials.

Recently, while researching for my project, I stumbled upon PyFluent and am intrigued by its potential applications in my dissertation. My aim is to explore the use of Machine Learning (ML) or genetic algorithms to optimize a specific component within a nuclear reactor.

Based on initial discussions with my university supervisor, I am considering a topic along the lines of: "Utilization of Machine Learning for Heat Exchanger Optimization". However, I'm still in the process of refining the specifics, such as which component to focus on and what parameters to optimize.

My primary question to this community is: **Is it feasible to implement ML or genetic algorithms within the PyAnsys framework, specifically with PyFluent?** If so, could anyone point me in the direction of someone knowledgeable in this area with whom I can discuss the technicalities?

Furthermore, I would greatly appreciate any feedback, comments, or insights regarding my dissertation proposal. If there are any ongoing research projects or studies that align with my interests, I would love to learn more and potentially collaborate.

Thank you in advance for your time and assistance. I eagerly await your responses and insights.

Kind regards, 
Harry Burgess
HBurgess173.HB@gmail.com

**Comment by RobPasMue** [comment]
Hi @Crypto-Aitch!

Glad you found us! Let me try to give you a bit of context. PyAnsys is an Ansys initiative to contribute to open-source through Pythonic libraries that wrap around Ansys products (as well as other more general tools that can be used by any Python projects!). They will enable the use of Ansys products from a Python terminal and it will allow you to interact with your results from whatever other Python library you prefer!

The use of ML has been explored by some of the libraries such as PyMAPDL. See an example repository here https://github.com/ansys/ml-rl-cartpole and its documentation https://cartpole.mapdl.docs.pyansys.com/

A similar approach could be followed using PyFluent for sure! It's just a matter of evaluating your simulation results with the ML Python library of your choice and iterating on your simulation parameters! Seems feaasible at first sight.

Let me ping the PyFluent maintainers and AFT group in case they are interested on helping you out! @ansys/pyfluent-aft @ansys/pyfluent-contributors @ansys/pyfluent-maintainers

**Comment by hpohekar** [comment]
Hi @Crypto-Aitch , @RobPasMue 

Following example demonstrates Machine Learning model building using PyFluent.

https://examples.fluent.docs.pyansys.com/version/dev/examples/00-released_examples/02-DOE_ML.html#sphx-glr-examples-00-released-examples-02-doe-ml-py

Have a look. 

Thank you.
